rollbar-reactive-streams/src/test/java/com/rollbar/reactivestreams/notifier/sender/http/AsyncHttpClientFactoryTest.java

```diff
@@ -0,0 +1,13 @@
+package com.rollbar.reactivestreams.notifier.sender.http;
+
+import static org.junit.Assert.assertThat;
+
+import org.hamcrest.Matchers;
+import org.junit.Test;
+
+public class AsyncHttpClientFactoryTest {
+  @Test
+  public void ifApacheIsInClasspathItShouldReturnApacheClient() {
+    assertThat(AsyncHttpClientFactory.defaultClient(), Matchers.instanceOf(ApacheAsyncHttpClient.class));
+  }
```

Comment:
Instead of renaming test, how about adding a negative test?

What is expected behaviour for the factory if the Apache HTTP Client library is not available? Do you return void, throw an exception? Whatever you do in this case, please document it by adding a test for this behaviour.
This will solve issue with too abbreviated name.

Reply:
Adding a negative test is clearly the standard approach here, and I'd like to do it, but the effort is pretty large because classpath setup is not available within the test itself, and it has to be done externally. It involves either:

1. a custom test runner and classloader, which is a lot of code
2. setting up two different gradle classpath configurations and running two different test tasks

So not including the negative test is a compromise that I think is acceptable here given the circumstances.

Reply:
Agreed. Thanks for the descriptive comments!